sdk/identity/azure-identity/src/private/token_cache_internals.hpp

```diff
@@ -76,13 +80,13 @@ namespace Azure { namespace Identity { namespace _detail {
      * A test hook that gets invoked before cache write lock gets acquired.
      *
      */
-    static std::function<void()> OnBeforeCacheWriteLock;
+    AZ_IDENTITY_DLLEXPORT static std::function<void()> OnBeforeCacheWriteLock;
```

Comment:
Why use AZ_IDENTITY_DLLEXPORT here? That only makes sense if you're building a DLL on Windows, and I believe that all our code consumes the SDK as a static library.

Reply:
I believe you can build it any way you want - you can build as DLL and then the tests would be consuming it as DLL.

Reply:
External customers can build the SDK as a DLL. But as best as I've been able to figure out, our internal test infrastructure never uses the SDK as a DLL (it's one of the major weaknesses of our current tests).

Reply:
I would not be supporting this assumption then, especially if we think it is a problem. In the SDK code, we do have to put the DLLEXPORT attribute before all the static fields, if they end up being used by customer, or by inline SDK functions, I don't see the reason to break the pattern.

Reply:
Btw, this also means that this is a part of the public API contract for this type (because you are exporting it from the DLL - all DLL exports are a part of the public API contract, whether they're in an internal namespace or not).

Reply:
Technically all the stuff in `_internal` or `_detail` namespaces is visible, it only is a matter of agreement between us and the SDK users so that SDK users don't use them.